src/compiler.rs

```diff
@@ -275,28 +255,26 @@ impl<CM: CellManager, SSB: StepSelectorBuilder> Compiler<CM, SSB> {
             None
         };
 
-        Circuit::<F, TraceArgs, StepArgs> {
-            placement: unit.placement,
-            selector: unit.selector,
-            columns: unit.columns,
-            exposed: unit.exposed,
-            polys: unit.polys,
-            lookups: unit.lookups,
-            step_types: unit.step_types,
-            q_enable,
-            q_first,
-            q_last,
-
-            trace: sc.trace.as_ref().map(|v| Rc::clone(v)),
-            fixed_gen: sc.fixed_gen.as_ref().map(|v| Rc::clone(v)),
-        }
+        (
+            Circuit::<F> {
+                placement: unit.placement,
+                selector: unit.selector,
+                columns: unit.columns,
+                exposed: unit.exposed,
+                polys: unit.polys,
+                lookups: unit.lookups,
+                step_types: unit.step_types,
+                q_enable,
+                q_first,
+                q_last,
+
+                fixed_gen: sc.fixed_gen.as_ref().map(|v| Rc::clone(v)),
+            },
+            sc.trace.as_ref().map(|v| TraceGenerator::new(Rc::clone(v))),
+        )
```

Comment:
What's the purpose of separating out `trace` while previously it's a part of the ir::Circuit?

Reply:
As before